server/xpub-server/routes.js

```diff
@@ -0,0 +1,67 @@
+const logger = require('@pubsweet/logger')
+const config = require('config')
+const { dbExists } = require('@pubsweet/db-manager')
+const AWS = require('aws-sdk')
+
+const FILE_HEALTH_CHECK = 'healthcheck-file'
+const DATABASE_ERROR = 'Database Error'
+const S3_CONNECTION_ERROR = 'S3 Connection Error'
+const DEFAULT_ERROR = 'Internal Server Error'
+const SUCCESFULL_RESPONSE = 'pong'
+const S3_API_VERSION = '2006-03-01'
+
+const nocache = (req, res, next) => {
+  res.header('Cache-Control', 'private, no-cache, no-store, must-revalidate')
+  res.header('Expires', '-1')
+  res.header('Pragma', 'no-cache')
+  next()
+}
+
+const s3 = new AWS.S3({
+  ...config.aws.credentials,
+  ...config.aws.s3,
+  apiVersion: S3_API_VERSION,
+})
+
+let lastHealthStatus = null
+
+module.exports = app => {
+  app.get('/ping', nocache, async (req, res) => {
+    const errors = []
+    let statusCode = 200
+    let body = SUCCESFULL_RESPONSE
+    let response
+    try {
+      response = await Promise.all([
+        dbExists(),
+        s3.getObject({ Key: FILE_HEALTH_CHECK }).promise(),
```

Comment:
We may as well `putObject` so we don't have to manually add the file to S3. And then run `getObject` afterwards, just as a sanity check.

Reply:
not sure if this is a good idea, the ping endpoint is being hit every 5 min, so putting an object will require to delete the object as well, maybe we are overloading this endpoint once actually what want is just check if we have connectivity

Reply:
5 _seconds_

Reply:
I would favor the database check as it's an absolute dependency, but I'd skip the S3 one in favor of `/ping` working reliably across environments.

Reply:
The S3 client will let you overwrite silently by default, but I agree with your point about keeping the response times low on this endpoint.

Reply:
Yes - let's keep this as-is